hera_sim/visibilities/simulators.py

```diff
@@ -340,7 +343,29 @@ def uvdata(self) -> UVData:
 
 
 class VisibilitySimulator(metaclass=ABCMeta):
-    """Base class for all hera_sim compatible visibility simulators."""
+    """Base class for all hera_sim-compatible visibility simulators.
+
+    To define a new simulator, make a subclass. The subclass should overwrite available
+    class-attributes as necessary, and specify a ``__version__`` of the simulator code
+    itself.
+
+    The :meth:`simulate` abstract method *must* be overwritten in the subclass, to
+    perform the actual simulation. The :meth::`validate` method *may* also be
+    overwritten to validate the given `UVData` input for the particular simulator.
+
+    The subclass may define any number of simulator-specific parameters as part of its
+    init method.
+
+    Finally, to enable constructing the simulator in command-line applications, a
+    :meth::`from_yaml` method is provided. This will load a YAML file's contents as a
+    dictionary, and then instantiate the subclass with the parameters in that dict.
+    To enable some control over this process, the subclass can overwrite the
+    :meth::`_from_yaml_dict` private method, which takes in the dictionary read from the
+    YAML file, and transforms any necessary parameters before constructing the class.
+    For example, if the class required a set of data from a file, the YAML might contain
+    the filename itself, and in :meth::`_from_yaml_dict`, the file would be read and the
```

Comment:
A few instances where `:meth::` needs to be replaced with `:meth:`

Reply:
Thanks for picking that up, fixed.